portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,17 +35,32 @@
 public final class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment");
-
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    String filter = request.getParameter("filter");
+
+    if (filter.equals("newest")) {
```

Comment:
There's a neat trick in Java where you can write: `"newest".equals(filter)` which returns `false` even in case filter happens to be null.